src/android/GoogleFitService.java

```diff
@@ -152,11 +154,21 @@ public synchronized Status insertData(DataSet dataSet) {
             for (DataSet dataSet : dataSets) {
 
                 for (DataPoint dp : dataSet.getDataPoints()) {
-                    if (dp.getOriginalDataSource().getAppPackageName() == null) {
+                    String appPkgName = dp.getOriginalDataSource().getAppPackageName();
+
+                    if (appPkgName == null) {
                         continue bucketLoop;
                     }
 
-                    activity.setSourceName(dp.getOriginalDataSource().getAppPackageName());
+                    PackageManager pm = this.activityContext.getPackageManager();
```

Comment:
[nice to have] tą deklaracje PackageManager najlepiej by było wrzucić przed pętlę bucketLoop (do linii 148) inaczej kilka razy będzie od nowa definiowane